src/components/Inputs/ObjectList.js

```diff
@@ -0,0 +1,92 @@
+import { FormSpy, useForm } from 'react-final-form'
+import { useDispatch, useSelector } from 'react-redux'
+import React, { useEffect } from 'react'
+import i18n from '@dhis2/d2-i18n'
+
+import { Field } from '../Field/Field'
+import { Label } from '../Field/Label'
+import { Select } from '../FinalFormComponents/Select'
+import { fetchObjects } from '../../reducers/object/thunks'
+import {
+    getObjects,
+    getObjectsError,
+    getObjectsLoading,
+} from '../../reducers/object/selectors'
+
+export const OBJECT_DEFAULT_OPTIONS = []
+export const OBJECT_KEY = 'objectList'
+export const OBJECT_DEFAULT_VALUE = null
+
+const objectLabel = i18n.t('Object')
+
+const RenderComponent = ({ values }) => {
```

Comment:
I'm still not really too familiar with hooks yet, is it normal to use components just for their side-effects like this? Seems kind of unnatural to me, especially with the name. Took me a bit to understand what's going on and why it's done this way. 
The onChange-callback on FromSpy would be more suiting I think, but then there would be more code in ObjectList.

Reply:
To my understanding there's a reason for this, as when I tried to replace with a simple callback, it will re-render and "forget" about the selected value. So I guess this is a way to (re)load the values on first render and when the objectType changes?

Reply:
Completely changed. I've send you a draft on slack before, but please have a look again